core/autoscaling.go

```diff
@@ -505,89 +647,299 @@ func (a *autoScalingGroup) getAnySpotInstance() *instance {
 	return a.getInstance(nil, false, false)
 }
 
-// returns an instance ID as *string and a bool that tells us if  we need to
-// wait for the next run in case there are spot instances still being launched
-func (a *autoScalingGroup) havingReadyToAttachSpotInstance() (*string, bool) {
+func isInstanceNotFound(err error) bool {
+	if aerr, ok := err.(awserr.Error); ok {
+		if aerr.Code() == "InvalidInstanceID.NotFound" {
+			return true
+		}
+	}
+	return false
+}
+
+func (a *autoScalingGroup) getInstanceState(instanceID *string) int64 {
+	svc := a.region.services.ec2
+	input := &ec2.DescribeInstanceStatusInput{
+		InstanceIds: []*string{
+			instanceID,
+		},
+		IncludeAllInstances: aws.Bool(true),
```

Comment:
Why including this parameter?

Reply:
The default param value is false.  Meaning it only returns healthy instances, and not terminated ones.  We need to know the terminated ones in case AWS can culled the instances associated to the SIR